katnip/targets/pykd_dbg.py

```diff
@@ -0,0 +1,295 @@
+                except Exception as err:
+                    self.logger.debug("Get system id fail because of: %s" % err)
+                    continue
+            # Set Bp
+            if self._wait_break():
+                self.logger.info("Server is in break status setting break points")
+                for bp in self._break_points:
+                    pykd.setBp(bp)
+                self.logger.info("Start register event handle")
+                # this will register our handle
+                self.handler = self._handler(self)
+                self.logger.info('Handler object is : %s' % self.handler)
+                self.logger.info('Go !!!!!')
+                pykd.go()
+        except:
+            self.logger.error('Got an exception in _debug_server')
+            self.logger.error(traceback.format_exc())
+
+    def _start_server_thread(self):
+        '''
+        start the server thread
+        '''
+        self.server_thread = FuncThread(self._debug_server)
+        self.server_thread.start()
+
+    def _kill_all_processes(self):
+        '''
+        kill all processes with the same name
+        :return: True if all matching processes were killed properly, False otherwise
+        '''
+        if self._process:
+            try:
+                status = self._get_exe_status()
+                self.logger.info("Current status is %s start kill process" % status)
+                if status == 'Break':
+                    self.logger.info("Process is in break status, kill process now")
+                    pykd.killAllProcesses()
+                    self._pid = None
+                    self._process = None
+                else:
+                    self.logger.info("Break in process, kill process now")
+                    pykd.breakin()
+                    self._wait_break()
+                    # TODO: need find a way to replace time.sleep
+                    time.sleep(0.05)
+                    pykd.killAllProcesses()
+                    self._pid = None
+                    self._process = None
+            except:
+                self.logger.error('failed to kill process [%s]' % traceback.format_exc())
+                return False
+            return True
+
+    def teardown(self):
+        self._stop_process()
+        self._pid = None
+        self._process = None
+        super(PykdTarget, self).teardown()
+
+    def pre_test(self, test_number):
+        super(PykdTarget, self).pre_test(test_number)
+        self.is_crash.clear()
+        self.crash_dump_finished.clear()
+        self._stop_process()
+
+    def post_test(self, test_num):
+        if self.is_crash.wait(self._timeout):
+            self.crash_dump_finished.wait(self._timeout)
+        self._stop_process()
+        # for each test case we need deinitialize the pykd, other wise event handle will not work.
+        pykd.deinitialize()
+        super(PykdTarget, self).post_test(test_num)
+
+    def _send_to_target(self, data):
+        '''
```

Comment:
mutational_data is not a got description, why not use the original description?
typo - might by --> might be